File: dotnet/Library/Okapi/OkapiMetadata.cs
using Okapi.Proofs.V1;

namespace Okapi.Metadata;

public static class OkapiMetadata
{
    /// <summary>
    ///     Generate new key
    /// </summary>
    /// <param name="request"></param>
    /// <returns></returns>
    public static MetadataResponse GetMetadata()
    {
        return Native.Call<MetadataRequest, MetadataResponse>(new MetadataRequest(), Native.okapi_metadata);
    }
}

File: java/src/main/java/trinsic/okapi/OkapiMetadata.java
package trinsic.okapi;

import com.google.protobuf.InvalidProtocolBufferException;
import trinsic.okapi.metadata.Metadata;

public class OkapiMetadata extends OkapiNative {
  public static Metadata.MetadataResponse getMetadata()
      throws DidException, InvalidProtocolBufferException {
    var request = Metadata.MetadataRequest.getDefaultInstance();
    OkapiByteBuffer.ByValue requestBuffer = messageToBuffer(request);
    OkapiByteBuffer responseBuffer = new OkapiByteBuffer();
    ExternError errBuffer = new ExternError();
    var result = getNativeLibrary().oberon_create_key(requestBuffer, responseBuffer, errBuffer);
    errBuffer.raiseError(result);
    return Metadata.MetadataResponse.parseFrom(bufferToByteArray(responseBuffer));
  }
}
